agent/engine/dockerclient/dockerclientfactory_windows_test.go

```diff
@@ -53,6 +53,18 @@ func TestFindClientAPIVersion(t *testing.T) {
 	for _, version := range getAgentVersions() {
```

Comment:
I'd split TestFindClientAPIVersion into two different tests, since the assertions are different.

One that iterates over getWindowsReplaceableVersions(), and one that iterates over getWindowsSupportedVersions().

Reply:
This test gets clients for all the agent supported versions i.e `getAgentVersions()` (this now includes replaceable and supported versions), hence extended this for `getWindowsSupportedVersions()` too.